pkg/flow/adapter/dataweavetransformation/adapter.go

```diff
@@ -96,49 +115,97 @@ func (a *dataweaveTransformAdapter) Start(ctx context.Context) error {
 
 func (a *dataweaveTransformAdapter) dispatch(ctx context.Context, event cloudevents.Event) (*cloudevents.Event, cloudevents.Result) {
 	var err error
-	var tmpfile *os.File
-	path := "/tmp/dw"
-	dwFolder := path + "/custom"
+	var tmpFile *os.File
+	var inputData []byte
+	var spell string
+	var inputContentType string
+	var outputContentType string
 
-	switch a.incomingContentType {
-	case "application/json":
-		if !isValidJSON(event.Data()) {
-			return a.replier.Error(&event, targetce.ErrorCodeRequestParsing,
-				errors.New("invalid Json"), nil)
+	err = validateContentType(event.DataContentType(), event.Data())
+	if err != nil {
+		return a.replier.Error(&event, targetce.ErrorCodeRequestValidation,
+			errors.New(err.Error()), nil)
+	}
+
+	req := &DataWeaveTransformationStructuredRequest{}
+	if err := event.DataAs(req); err != nil {
+		return a.replier.Error(&event, targetce.ErrorCodeRequestParsing, err, nil)
+	}
+
+	if !a.spellOverride && req.Spell != "" {
+		return a.replier.Error(&event, targetce.ErrorCodeRequestValidation,
+			errors.New("it is not allowed to override Spell per CloudEvent"), nil)
+	}
+
+	if a.defaultSpell == nil && req.Spell == "" {
+		return a.replier.Error(&event, targetce.ErrorCodeRequestValidation,
+			errors.New("no default Spell or in request found"), nil)
+	}
+
+	if req.InputContentType != "" || req.OutputContentType != "" || req.Spell != "" {
+		if req.InputData == "" {
+			return a.replier.Error(&event, targetce.ErrorCodeRequestValidation,
+				errors.New("inputData not found"), nil)
 		}
-		tmpfile, err = ioutil.TempFile(path, "*.json")
+	}
```

Comment:
These checks seem a bit clunky to me as you could rely on the defaults specified by the adapter when checking if the associated attributes are not included with the payload.